src/catalog/catalog_cache.cpp

```diff
@@ -157,5 +159,76 @@ std::shared_ptr<IndexCatalogObject> CatalogCache::GetCachedIndexObject(
   return nullptr;
 }
 
+/*@brief   insert sequence catalog object into cache
+ * @param   sequence_object
+ * @return  false only if sequence already exists in cache
+ */
+bool CatalogCache::InsertSequenceObject(
+    std::shared_ptr<SequenceCatalogObject> sequence_object) {
+  if (!sequence_object || sequence_object->seq_oid == INVALID_OID) {
+    return false;  // invalid object
+  }
+
+  std::size_t hash_key = GetHashKey(sequence_object->seq_name,
+                                    sequence_object->db_oid);
+
+  // check if already in cache
+  if (sequence_objects_cache.find(hash_key) !=
+          sequence_objects_cache.end()) {
+    LOG_DEBUG("Sequence %s already exists in cache!",
+              sequence_object->seq_name.c_str());
+    return false;
+  }
+
+  sequence_objects_cache.insert(
+          std::make_pair(hash_key, sequence_object));
+  return true;
+}
+
+/*@brief   evict sequence catalog object from cache
+ * @param   sequence_name, database_oid
+ * @return  true if specified sequence is found and evicted;
+ *          false if not found
+ */
+bool CatalogCache::EvictSequenceObject(const std::string & sequence_name,
+         oid_t database_oid) {
+  std::size_t hash_key = GetHashKey(sequence_name, database_oid);
+
+  auto it = sequence_objects_cache.find(hash_key);
+  if (it == sequence_objects_cache.end()) {
+    return false;  // sequence not found in cache
+  }
+
+  auto sequence_object = it->second;
+  PELOTON_ASSERT(sequence_object);
+  sequence_objects_cache.erase(it);
+  return true;
+}
+
+/*@brief   get sequence catalog object from cache
+ * @param  sequence_name, database_oid
+ * @return  sequence catalog object; if not found return object with invalid oid
+ */
+std::shared_ptr<SequenceCatalogObject> CatalogCache::GetSequenceObject(
+            const std::string & sequence_name, oid_t database_oid) {
+  std::size_t hash_key = GetHashKey(sequence_name, database_oid);
+  auto it = sequence_objects_cache.find(hash_key);
+  if (it == sequence_objects_cache.end()) {
+    return nullptr;
```

Comment:
It said it return object with invalid oid if not found. To make things consistent, instead of returning nullptr, we should probably return an SequenceCatalogObject with seq_oid = INVALID_OID. Or another easier way is to change the spec.

Reply:
This is a consistent behavior with the existing methods (like GetTableObejct, etc.) in catalog_cache.cpp so I would like to keep it.